src/actions/LoginAction.tsx

```diff
@@ -56,13 +56,15 @@ export const launchPasswordRecovery = (recoveryKey: string) => async (
       })
       logEvent('Recovery_Username_Success')
       return true
-    } catch (error: any) {
-      showError(
-        error != null && error.name === 'UsernameError'
-          ? s.strings.recovery_by_username_error
-          : error
-      )
+    } catch (error: unknown) {
       logEvent('Recovery_Username_Failure')
+
+      const usernameError = asMaybeUsernameError(error)
+      if (usernameError != null) {
+        return s.strings.recovery_by_username_error
+      }
```

Comment:
This block returns the error as a string, so it appears inside the modal instead of as a drop-down. I think the experience is better this way.

Reply:
Nope, looks bad on smaller phones, so we have to revert:

![simulator_screenshot_708B43AD-E975-40D5-B231-A7C294DCF05B](https://github.com/EdgeApp/edge-login-ui-rn/assets/276214/5ef9fb0e-be70-4f73-a34a-8baccffa12ba)